src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs

```diff
@@ -59,6 +60,18 @@ private static unsafe SafeMsQuicConfigurationHandle Create(QuicOptions options,
                 throw new Exception("MaxBidirectionalStreams overflow.");
             }
 
+            if ((flags & QUIC_CREDENTIAL_FLAGS.CLIENT) == 0)
+            {
+                if (certificate == null)
+                {
+                    throw new Exception("Server must provide certificate");
+                }
+            }
+            else
+            {
+                flags |= QUIC_CREDENTIAL_FLAGS.INDICATE_CERTIFICATE_RECEIVED | QUIC_CREDENTIAL_FLAGS.NO_CERTIFICATE_VALIDATION;
```

Comment:
You never want to let the TLS stack handle default validation?

Reply:
That how SslStream works now. Long term, we may check if custom callback is present or now. But that is still problematic (at least on Linux) as X509Chain would consider user stores while OS does not have any clue about them. So we would end up in situation when SslStream works and Quick does not or vice versa.